android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/PatientExtension.kt

```diff
@@ -211,3 +212,14 @@ fun Patient.extractOfficialIdentifier(): String? =
   if (this.hasIdentifier())
     this.identifier.firstOrNull { it.use == Identifier.IdentifierUse.OFFICIAL }?.value
   else null
+
+fun Patient.extractTypeViaDTreeMeta(): PatientType {
+  return try {
+    val tagList = this.meta.tag.filter { it.system.equals("https://d-tree.org", true) }
+
+    PatientType.valueOf(tagList[0].code?.uppercase(Locale.getDefault())?.replace("-", "_") ?: "")
```

Comment:
This code is error prone. Empty string cannot be converted to `PatientType`

Reply:
fixed